lib/api/widget/functions.php

```diff
@@ -218,7 +214,7 @@ function beans_widget_area_shortcodes( $content ) {
 		$content = build_query( $content );
 	}
 
-	return beans_array_shortcodes( $string, $GLOBALS['_beans_widget_area'] );
+	return beans_array_shortcodes( $content, $GLOBALS['_beans_widget_area'] );
```

Comment:
Fixed a bug! Great job! 👍 

Question: Is it possible that the key `'_beans_widget_area'` might not exist in the global?  Could we get to here and that key is not in the superglobal?  If no, cool! If yes, protect it.

Reply:
It's a good question. This function isn't actually used by Beans itself. It seems to be a means for adding shortcode processing to widget areas by child themes. Under ordinary circumstances, I think it would only be called after a widget area has been set up (which sets the GLOBAL). However, since a creative themer may try to call it from elsewhere, it's probably better off protected.

Reply:
Same for `beans_widget_shortcodes()`, below.

Reply:
Right.  Now, we can address the _timing issue_ when we rearchitect the globals.  We'll eliminate all timing issues by storing it in the state container and then pulling from the state when it's time to use it.  We do that now in the Actions API.  We'll reuse that same design concept everywhere.  

I rather like the idea of getting rid of timing issues of registering or doing something too early.  Shoot, record what you want to do in the state manager and then let it handle it when its needed.

But for this release, we need to protect the code to help out the theme/plugin developer.